presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchMetadata.java

```diff
@@ -503,6 +508,28 @@ else if (predicatePushdownEnabled && handle.getTableName().equals(TpchTable.PART
                 unenforcedConstraint));
     }
 
+    @Override
+    public Optional<TableScanRedirectApplicationResult> applyTableScanRedirect(ConnectorSession session, ConnectorTableHandle table)
+    {
+        TpchTableHandle handle = (TpchTableHandle) table;
+        Optional<String> destinationCatalog = Optional.ofNullable(session.getProperty(TPCH_TABLE_SCAN_REDIRECTION_CATALOG, String.class));
+        Optional<String> destinationSchema = Optional.ofNullable(session.getProperty(TPCH_TABLE_SCAN_REDIRECTION_SCHEMA, String.class));
+        if (destinationCatalog.isEmpty() && destinationSchema.isEmpty()) {
+            return Optional.empty();
+        }
+
+        CatalogSchemaTableName destinationTable = new CatalogSchemaTableName(
+                destinationCatalog.orElse("tpch"),
```

Comment:
Let's skip redirect on missing as one doesn't know which catalog tpch connector will be mounted in.